pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -142,7 +143,10 @@ func (s *volumeStatCalculator) calcAndStoreStats() {
 		if err != nil {
 			// Expected for Volumes that don't support Metrics
 			if !volume.IsNotSupported(err) {
-				klog.V(4).InfoS("Failed to calculate volume metrics", "pod", klog.KObj(s.pod), "podUID", s.pod.UID, "volumeName", name, "err", err)
+				if oldErr, found := errors[name]; !found || oldErr.Error() != err.Error() {
```

Comment:
Should we clear the error on a success?

What happens if the pod gets deleted without it ever succeeding? Will we leak the entry in the cache?

Reply:
@msau42 

1. What is name? Is it unique across pods?
name is volumeName and it is unique to pod. 

2. Should we clear the error on a success?
Thanks for the good suggestion. I resolve your comments by delete name from `errors map[string]error` when s succeed happen.

3. What happens if the pod gets deleted without it ever succeeding? Will we leak the entry in the cache?
This question is not related to this PR. This PR only change the logging and did not touch anything related to cache and pod volume calculation.

Reply:
>> What is name? Is it unique across pods?
> name is volumeName and it is unique to pod.

I think concern here is that the same volume may be defined on multiple pods. If error happens on only one of them - second one will erase the error. So the error from the first will be recorded again

>> What happens if the pod gets deleted without it ever succeeding? Will we leak the entry in the cache?
> This question is not related to this PR. This PR only change the logging and did not touch anything related to cache and pod volume calculation.

It is vey much related as the errors map is effectively a cache that will never be cleaned up in described scenario.

Reply:
2 completely different pods can name their volumes the same. So we need to make sure the volume name used here is uniquely identifiable across different pods.

Reply:
@SergeyKanzhelev @msau42 

If you see the code [here](https://github.com/kubernetes/kubernetes/blob/3be3b86dd642a35a4727788162388a990461b434/pkg/kubelet/server/stats/fs_resource_analyzer.go#L78C1-L84C3), where we call and use `volumeStatCalculator`. This `volumeStatCalculator` is per pod basis, which means each pod has its own `volumeStatCalculator`. And we call `StartOnce()` to calculate the pod volume and `StopOnce()` to stop the calculation. So, there should not be a situation two pods share the same (errors map[string]error.